jsurfer-core/src/main/java/org/jsfr/json/compiler/JsonPathCompiler.java

```diff
@@ -297,14 +299,72 @@ public Void visitFilterMatchRegex(JsonPathParser.FilterMatchRegexContext ctx) {
         return rst;
     }
 
-    private static String removeQuote(String quotedString) {
-        return quotedString.substring(1, quotedString.length() - 1);
+    static String unescapeString(String quotedString) {
+        assert quotedString.startsWith("\"") && quotedString.endsWith("\"");
+        StringBuilder res = new StringBuilder(quotedString.length() - 2);
+        int end = quotedString.length() - 1;
+        for (int i = 1; i < end; i++) {
+            char ch = quotedString.charAt(i);
+            assert ch != '"' : "unexpected end quote";
+            if (ch == '\\') {
+                ch = quotedString.charAt(++i);
+                switch (ch) {
+                    case '"':
+                    case '\'':
+                    case '\\':
+                    case '/':
+                        break;
+
+                    case 'b':
+                        ch = '\b';
+                        break;
+
+                    case 'f':
+                        ch = '\f';
+                        break;
+
+                    case 'n':
+                        ch = '\n';
+                        break;
+
+                    case 'r':
+                        ch = '\r';
+                        break;
+
+                    case 't':
+                        ch = '\t';
+                        break;
+
+                    case 'u':
+                        if (i + 4 >= end) {
+                            throw new JsonSurfingException("Invalid escape");
```

Comment:
Maybe `JsonPathCompilerException` is more relevant here since this exception occurs on json path compilation stage.